#!/usr/bin/env python
# ftgr.cfg
"""Dynamic reconfigure config file."""
PACKAGE = "follow_the_gap-v0"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Speed_max", double_t, 0, "Level-1 speed", 0.158483034, 0, 0.3)
gen.add("Speed_min", double_t, 0, "Level-2 speed", 0.138522954, 0, 0.3)
gen.add("Speed_avoid", double_t, 0, "Level-3 speed", 0.098602794, 0, 0.3)

gen.add("TurnL_min", double_t, 0, "Level-1 turn ratio (left)", 0.677419355, 0, 2)
gen.add("TurnL_max", double_t, 0, "Level-2 turn ratio (left)", 1.000000000, 0, 2)
gen.add("TurnL_avoid", double_t, 0, "Level-3 turn ratio (left)", 1.419354839, 0, 2)

gen.add("TurnR_min", double_t, 0, "Level-1 turn ratio (right)", 0.677419355, 0, 2)
gen.add("TurnR_max", double_t, 0, "Level-2 turn ratio (right)", 0.709677419, 0, 2)
gen.add("TurnR_avoid", double_t, 0, "Level-3 turn ratio (right)", 1.290322581, 0, 2)

gen.add("Switch_L12", double_t, 0, "Level-1 to Level-2 angle switch", 20, 0, 90)
gen.add("Hysteresis_L12", double_t, 0, "Level-1 to Level-2 angle hysteresis", 5, 0, 20)

gen.add("Switch_L23", double_t, 0, "Level-2 to Level-3 angle switch", 41, 0, 90)
gen.add("Hysteresis_L23", double_t, 0, "Level-2 to Level-3 angle hysteresis", 5, 0, 20)

exit(gen.generate(PACKAGE, "follow_the_gap_ride", "ftgr"))
